src/Http/ResponseMatcher/ResponseMatcher.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCacheBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCacheBundle\Http\ResponseMatcher;
+
+use Symfony\Component\DependencyInjection\ExpressionLanguage;
+use Symfony\Component\HttpFoundation\Response;
+
+class ResponseMatcher implements ResponseMatcherInterface
+{
+    private $statusCodes;
+    private $expression;
+    private $expressionLanguage;
+
+    public function matchExpression($expression)
```

Comment:
i would prefer to call this setMatchExpression to make it explicit this is setting up the matcher. (and same for status code below). or have a constructor so that we can require at least one of the two criteria to exist? 

the phpdoc for $expression should explain what context the expression gets.